src/connection.js

```diff
@@ -1672,13 +1718,8 @@ class Connection extends EventEmitter {
       const message = 'Requests can only be canceled in the ' + this.STATE.SENT_CLIENT_REQUEST.name + ' state, not the ' + this.state.name + ' state';
```

Comment:
When cancel() is called after a timeout, a debug message is still displayed, see #545.

Reply:
I want to deprecate `.cancel` on `Connection`, and thus did not touch this function more than necessary. We could change this function to:

```js
if (!this.request) {
  return false;
}

if (this.request.canceled) {
  return false;
}

this.request.cancel();
return true;
```

And still deprecate (and later remove) this function? 🤔